models/raydium/raydium_v3_trades.sql

```diff
@@ -0,0 +1,162 @@
+                else sp.call_outer_executing_account
+                end as trade_source
+            ,case
+                when lower(tokenA_symbol) > lower(tokenB_symbol) then concat(tokenB_symbol, '-', tokenA_symbol)
+                else concat(tokenA_symbol, '-', tokenB_symbol)
+            end as token_pair
+            , case when tk_1.token_mint_address = p.tokenA then COALESCE(tokenB_symbol, tokenB) 
+                else COALESCE(tokenA_symbol, tokenA)
+                end as token_bought_symbol 
+            -- token bought is always the second instruction (transfer) in the inner instructions
+            , tr_2.amount as token_bought_amount_raw
+            , tr_2.amount/pow(10,COALESCE(case when tk_1.token_mint_address = p.tokenA then p.tokenB_decimals else tokenA_decimals end,9)) as token_bought_amount
+            , case when tk_1.token_mint_address = p.tokenA  then COALESCE(tokenA_symbol, tokenA)
+                else COALESCE(tokenB_symbol, tokenB)
+                end as token_sold_symbol
+            , tr_1.amount as token_sold_amount_raw
+            , tr_1.amount/pow(10,COALESCE(case when tk_1.token_mint_address = p.tokenA then p.tokenA_decimals else tokenB_decimals end,9)) as token_sold_amount
+            , p.pool_id
+            , sp.call_tx_signer as trader_id
+            , sp.call_tx_id as tx_id
+            , sp.call_outer_instruction_index as outer_instruction_index
+            , COALESCE(sp.call_inner_instruction_index, 0) as inner_instruction_index
+            , sp.call_tx_index as tx_index
+            , case when tk_1.token_mint_address = p.tokenA then p.tokenB
+                else p.tokenA
+                end as token_bought_mint_address
+            , case when tk_1.token_mint_address = p.tokenA then p.tokenA 
+                else p.tokenB
+                end as token_sold_mint_address
+            , case when tk_1.token_mint_address = p.tokenA then p.tokenBVault
+                else p.tokenAVault
+                end as token_bought_vault
+            , case when tk_1.token_mint_address = p.tokenA then p.tokenAVault 
+                else p.tokenBVault
+                end as token_sold_vault
+        FROM {{ source('raydium_clmm_solana', 'amm_v3_call_swap') }} sp
+        INNER JOIN pools p
+            ON sp.account_poolState = p.pool_id --account 2
+            and p.recent_init = 1 --for some reason, some pools get created twice.
+        INNER JOIN {{ source('spl_token_solana', 'spl_token_call_transfer') }} tr_1 
+            ON tr_1.call_tx_id = sp.call_tx_id 
+            AND tr_1.call_outer_instruction_index = sp.call_outer_instruction_index 
+            AND ((sp.call_is_inner = false AND tr_1.call_inner_instruction_index = 1) 
+                OR (sp.call_is_inner = true AND tr_1.call_inner_instruction_index = sp.call_inner_instruction_index + 1))
+            {% if is_incremental() %}
+            WHERE {{incremental_predicate('tr_1.call_block_time')}}
+            {% else %}
```

Comment:
@jeff-dude @aalan3 how do I pass a table alias into this predicate? this does not seem to be working

Reply:
`WHERE` -> `AND` you are still inside the join here :)

Reply:
ah yeah, it's failing on other models like Phoenix too though